src/com/palmergames/bukkit/towny/object/PlotGroup.java

```diff
@@ -93,29 +95,28 @@ public Resident getResident() throws NotRegisteredException {
 
 	public boolean hasResident() { return resident != null; }
 	
-	public void addTownBlock(TownBlock townBlock) {
-		if (townBlocks == null)
-			townBlocks = new ArrayList<>();
+	public boolean addTownBlock(@NotNull TownBlock townBlock) {
+		if (townBlocks.containsValue(townBlock)) {
```

Comment:
I understand you're trying to make use of the `containsValue` optimization, but the implementation detail means that this will call `getWorldCoord()` on the townblock anyway which will create a new `WorldCoord` object. 
That object should be stored to a variable and used rather than having to create two of them. 

Also refer to above where I mentioned that these types of function can be changed to one-liners.